### ISSUE TITLE
Using Sequence as validation data in Model.fit

### ISSUE THREAD

**Issue by z1103246** [tensorflow/tensorflow#39797]
According to the documentation, Model.fit_generator is deprecated now in favor of Model.fit, which now supports Sequence and generators. However, the doc also states that Sequence object cannot be used for the validation set. Is there any plan to add support to enable the use of Sequence or other custom generators in the future?

**Comment by jessicasena** [comment]
This would be a very useful feature. I think that is counterintuitive to use your `x` data as `tf.keras.Sequence` and your `validation_data` as a `tuple` or `tf.data.Dataset`. And,  I didn't find any tutorial of how to porting codes that use`tf.keras.Sequence` to `tf.data.Dataset`, and it does not seem to be easy enough to justify the lack of a guide about how to do this migration. If `tf.keras.Sequence` is not deprecated, why `validation_data` argument in `tf.keras.fit()` cannot support it?

**Comment by fshofmann** [comment]
I havent tested it yet (will do soon) but what further confuses me regarding this issue is the bit about the` validation_batch_size` parameter found in the same documentation that tells us that `validation_data` does not support `Sequence`.
https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit

> 
Integer or None. Number of samples per validation batch. If unspecified, will default to batch_size. Do not specify the validation_batch_size if your data is in the form of datasets, generators**, or keras.utils.Sequence instances (since they generate batches).**
--

Must be a copy paste error, because there is also this issue which begs for a fix: https://github.com/tensorflow/tensorflow/issues/35879 
Extremely annoying, especially now that `fit_generator `is deprecated.

**Comment by fshofmann** [comment]
> And, I didn't find any tutorial of how to porting codes that use tf.keras.Sequence to tf.data.Dataset, and it does not seem to be easy enough to justify the lack of a guide about how to do this migration.

I deleted my former response here because I checked the code since then. Validator does no longer support Sequence, that is indeed true. However there is a quick and easy solution to the port problem from Sequence to Dataset API for the Validator set, although in my eye it is not a clean one, because you return to Datasets and their disadvantages (also it might fail for some use cases): 

- Take `tf.keras.Sequence` subclass
- Create a new `tf.Dataset` by using [`tf.data.Dataset.from_generator`](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#from_generator). `tf.keras.Sequence` has implemented the `__iter__` override class method, so you can simply pass that (`my_sequence.__iter__`) as your generator argument. 

Don't forget to use the `dataset.batch()` method afterwards to split your data in batches!

**Comment by qlzh727** [comment]
I think we have implemented the support for all the data types, and you can use Sequence for validation_data as well. Basically training data and validation data have the same set of supported types.

The docstring will be updated in https://github.com/tensorflow/tensorflow/pull/38979. Closing this bug since it working as intended.
